src/mutation_data.cpp

```diff
@@ -4,10 +4,22 @@
 #include "enums.h" // tripoint
 #include "bodypart.h"
 #include "debug.h"
+#include "trait_group.h"
 #include "translations.h"
 
-#include <vector>
+#include <set>
 #include <map>
+#include <vector>
+
+typedef std::map<Trait_group_tag, Trait_creation_data *> TraitGroupMap;
+typedef std::set<trait_id> TraitSet;
+
+TraitSet trait_blacklist;
+TraitGroupMap trait_groups = {
+    // An empty dummy group, it will not generate any traits. However, it makes that trait group
+    // id valid, so it can be used all over the place without need to explicitly check for it.
+    {"EMPTY_GROUP", new Trait_group_collection(100)}
```

Comment:
Ugh. Raw owning pointers? Please don't do this. If you absolutely need pointers, wrap them in a smart pointer class (for example `unique_ptr`). Do you actually need a pointer here? Note that the values in a `std::map` have stable addresses, inserting or removing entries does not affect the address of the remaining values.